src/marks/text.js

```diff
@@ -105,3 +105,39 @@ function applyDirectTextStyles(selection, mark) {
   applyAttr(selection, "dx", mark.dx);
   applyAttr(selection, "dy", mark.dy);
 }
+
+// https://developer.mozilla.org/en-US/docs/Web/CSS/font-size
+const fontSizes = new Set([
+  // global keywords
+  "inherit",
+  "initial",
+  "revert",
+  "unset",
+  // absolute keywords
+  "xx-small",
+  "x-small",
+  "small",
+  "medium",
+  "large",
+  "x-large",
+  "xx-large",
+  "xxx-large",
+  // relative keywords
+  "larger",
+  "smaller"
+]);
+
+// The font size may be expressed as a constant in the following forms:
+// - number in pixels
+// - string keyword: see above
+// - string <length>: e.g., "12px"
+// - string <percentage>: e.g., "80%"
+// Anything else is assumed to be a channel definition.
+function maybeFontSizeChannel(fontSize) {
+  if (fontSize == null || typeof fontSize === "number") return [undefined, fontSize];
+  if (typeof fontSize !== "string") return [fontSize, undefined];
+  fontSize = fontSize.trim().toLowerCase();
+  return fontSizes.has(fontSize) || /^[+-]?\d*\.?\d+([eE][+-]?\d+)?(\w*|%)$/.test(fontSize)
```

Comment:
```suggestion
  return fontSizes.has(fontSize) || /^[+-]?\d*\.?\d+(e[+-]?\d+)?(\w*|%)$/.test(fontSize)
```

e is already lowercased (note: I wouldn't have thought to check the scientific notation form for numbers!)

Reply:
I found this helpful:

https://www.w3.org/TR/css-syntax-3/#number-token-diagram
https://drafts.csswg.org/css-values/#numbers